akka-actor/src/main/scala/akka/io/Dns.scala

```diff
@@ -20,7 +20,7 @@ abstract class Dns {
    * Lookup if a DNS resolved is cached. The exact behavior of caching will depend on
    * the akka.actor.io.dns.resolver that is configured.
    */
-  def cached(name: String): Option[Dns.Resolved] = None
+  def cached(name: String): Option[Dns.Resolved] = name match { case _ ⇒ None } // avoid compiler warning
```

Comment:
This looks kind of weird... I see no obvious better way though.

Reply:
Is this a useful compiler warning for non final classes? Should we just turn it off `-Ywarn-unused:-params`

Reply:
IMO turning off a `-Y*` is a bad idea

Reply:
Here I think the warning does not help, and yes it does look weird but I'd err on the side of the benefit outweighs?

Reply:
This is a papercut but I would recommend the workarounds over turning off all unused parameter linting.

This workaround is interesting, I've never seen it before!  But it does produce more bytecode (I checked).

The alternative I know (from https://github.com/scala/bug/issues/10347#issuecomment-305546337) is:

```suggestion
  def cached(@deprecated("unused", "") name: String): Option[Dns.Resolved] = None
```

Reply:
@dwijnand I am in full agreement on linting. I found @deprecated("unused", "") and a related scala bug ticket and gave it a try, it solves our problem, and safely.

Reply:
Then we should define:
```
package akka.util

final class unused extends deprecated("unused", "")
```

I have verified that it works and it doesn't leak out to Scaladoc as deprecated.

Reply:
I saw that work as well, I think it's ok as long as we accept it being sprinkled about only where needed. I think it's mainly on internal APIs but I am testing more precise compiler tunings to see if that is possible instead first. Less invasive. I can add the util if I find it is not.

Reply:
SGTM.

Reply:
`@deprecated` is on route for being `final` in 2.13 😕 https://github.com/scala/scala/pull/7449

Reply:
ouch

Reply:
I am off work now, let me fix it.

Reply:
@dwijnand @patriknw I have created an RP for reverting the change: refs https://github.com/scala/scala/pull/7527

Currently, there is no clear migration path, so we could defer this kind of change later.